src/AdaptiveSpinLock.cpp

```diff
@@ -6,9 +6,9 @@ namespace pipert {
 
 int AdaptiveSpinLock::spins_before_yield_ = 0;
 
-AdaptiveSpinLock::AdaptiveSpinLock() {
+AdaptiveSpinLock::AdaptiveSpinLock(const int spins_before_yield) {
   if (!spins_before_yield_) {
-    spins_before_yield_ = ((std::thread::hardware_concurrency() == 1) ? 1 : kSpinsBeforeYield);
+    spins_before_yield_ = ((std::thread::hardware_concurrency() == 1) ? 1 : spins_before_yield);
```

Comment:
I realized that this variable is static. So when you set a new cycle count, it will propagate immediately to all AdaptiveSpinlocks into the system. I think we should change this to a real member variable.

Reply:
Yes, it really makes more sense to have it be able tto be different among instances. Unless it was a clear point of the original design to have all spinlocks spin for the same length of time. I've changed it to a private member now, and documented the k-prefix version as the default value.